src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -933,9 +1673,594 @@ bool W3DDisplay::Is_LetterBox_Fading()
+                g_theNetwork->Get_Incoming_Bytes_Per_Second(),
+                g_theNetwork->Get_Incoming_Packets_Per_Second());
+            m_displayStrings[10]->Set_Text(str1);
+
+            str1.Format(L"OUT: %.2f bytes/sec, %.2f packets/sec",
+                g_theNetwork->Get_Outgoing_Bytes_Per_Second(),
+                g_theNetwork->Get_Outgoing_Packets_Per_Second());
+            m_displayStrings[11]->Set_Text(str1);
+
+            str1.Format(L"Run Ahead: %d, Net FPS: %d, Packet arrival cushion: %d",
+                g_theNetwork->Get_Run_Ahead(),
+                g_theNetwork->Get_Frame_Rate(),
+                g_theNetwork->Get_Packet_Arrival_Cushion());
+            m_displayStrings[12]->Set_Text(str1);
+
+            str1.Clear();
+            int count = g_theNetwork->Get_Num_Players();
+
+            for (int i = 0; i < count; i++) {
+                Utf16String str4;
+                str4.Format(L"%s: %d ", g_theNetwork->Get_Player_Name(i).Str(), g_theNetwork->Get_Slot_Average_FPS(i));
+                str1.Concat(str4);
+            }
+
+            m_displayStrings[13]->Set_Text(str1);
+        } else {
+            str1.Format(L"");
+            m_displayStrings[11]->Set_Text(str1);
+            m_displayStrings[10]->Set_Text(str1);
+            m_displayStrings[12]->Set_Text(str1);
+            m_displayStrings[13]->Set_Text(str1);
+        }
+
+        int count = g_theInGameUI->Get_Select_Count();
+        str1.Format(L"Select Info: '%d' drawables selected", count);
+
+        if (g_theInGameUI->Get_Select_Count() == 1) {
+            d = g_theInGameUI->Get_First_Selected_Drawable();
+        }
+
+        if (d) {
+            const Object *o = d->Get_Object();
+            Utf8String str4;
+            str4.Set("No-Name");
+
+            if (o) {
+                Utf8String name = o->Get_Name();
+
+                if (!name.Is_Empty()) {
+                    str4 = o->Get_Name();
+                }
+            }
+
+            str1.Format(L"Select Info: '%S'(%S) at (%.3f,%.3f,%.3f)",
+                d->Get_Template()->Get_Name().Str(),
+                str4.Str(),
+                d->Get_Position()->x,
+                d->Get_Position()->y,
+                d->Get_Position()->z);
+
+            const PhysicsBehavior *p = o->Get_Physics();
+
+            PhysicsTurningType turn;
+            if (p) {
+                turn = p->Get_Turning();
+            } else {
+                turn = TURN_NONE;
+            }
+
+            DrawableLocoInfo *info = d->Get_Loco_Info();
+            if (info) {
+                str2.Format(L"\nPhysics Info -- Turn: %d, Pitch(accel): %.3f(%.3f), Roll(accel): %.3f(%.3f)",
+                    turn,
+                    info->m_accelerationPitch,
+                    info->m_accelerationPitchRate,
+                    info->m_accelerationRoll,
+                    info->m_accelerationRollRate);
+                str1.Concat(str2);
+            }
+
+#ifdef GAME_DEBUG_STRUCTS
```

Comment:
Did you test with `GAME_DEBUG_STRUCTS` compiled in? We had a few compile errors recently because of such code.

Reply:
Yes I tested.